### ISSUE TITLE
Slow speed on client side on some vps servers

### ISSUE THREAD

**Issue by yakamasix** [Nyr/openvpn-install#372]
Hi,
I have big problem with some vps servers, it doesnt matter its ubuntu 12,14,16 or debian 6,7,8 try every, all have 1Gbit network, on windows vpn client i got only 2-5 Mbps or even less https://snag.gy/1hEJje.jpg left vps, right client.
How to resolve this? i read many tutorials, i beginner in linux, please write me an ideas

**Comment by yakamasix** [comment]
client have 80 Mbits connection, UDP protocol is using

**Comment by lukebarone** [comment]
This will likely get closed but I'll try to help anyways.

What VPS are you using? What tier have you paid for? What other services
are running? Where (geographically) are you located, vs your VPS?

**Comment by yakamasix** [comment]
Im from Poland, server located in ukraine (this ping is normally 60ms, dont know why was 220) i must using this server. Deltahost.com, paid 5,4$ month https://deltahost.com/vps.html linux server 1gb ram, no services are running, clear installation with apt-get upgrade apt-get update, nothing else, i try ubuntu, centos, debian, all versions, i ask them about limits, no limit i shoud get full speed, i read many topics in internet, people have the same problem in some servers with openvpn

**Comment by yakamasix** [comment]
i know that you see GUI screenshot, but later i reinstall and install this script on clear systems, doesnt help

**Comment by yakamasix** [comment]
im a noob in linux, try to run this but no result :( can you give me your skype? i can provide teamviewer ID and password and if you have time you check whats going on

**Comment by yakamasix** [comment]
i installed speedtest-cli, because i have clear system ubuntu 16 now with intalled openvpn script from this site, what now? i try run https://github.com/sivel/speedtest-cli try without run (sorry im a noob)

**Comment by yakamasix** [comment]
this is problem with openvpn, i have with them windows server 1008-2012 and PPTP work perfectly, the same ip address, so i know this is problem with connection to them via openvpn

**Comment by Nyr** [comment]
There are some important points:
- Server can have 1 gbps but you shouldn't expect that kind of speed to nearly any place.
- VPS has shared resources, this means that both the CPU and network port can perform poorly.
- Bandwidth and latency usually get worse when increasing distance from client to host.
- Not every protocol is the same. PPTP is generally speaking faster than OpenVPN.
- Speedtest.net is not a reliable way of measuring speed, but I understand that it is a quick and easy way to get a first impression.

That said:
- You are testing against a server in Texas, very far away from Ukraine. That speed/latency result is nothing unexpected.
- It's very common for Ukranian and Russian ISPs to have congested networks. Likely a factor too.

Coincidentally, I also use an Ukranian server for VPN. I am connecting to my server from Spain, but just so you get an idea, I did a test from mine so you can see:
![6715371657](https://user-images.githubusercontent.com/739489/31690572-62443c2e-b393-11e7-9697-33cc5af091c2.png)

**TL;DR:** the internet is complex, your VPN is working fine. If you want a faster connection, you'll need a server with other ISP.

**Comment by Nyr** [comment]
Please note that closing the issue just means that there isn't an issue with the project. Is not an order for collaborators to stop helping the affected user ;)

That said, please don't spam with multiple short messages. I don't think there is much more that we can do for you in this specific case.

**Comment by yakamasix** [comment]
so is difference between windows PPTP and linux openvpn?

**Comment by yakamasix** [comment]
why i got 60 mbps on windows? one week ago i decide to migrate from windows to linux beacause is cheaper, and i think that will be the same speed similar 60mbps, not 10 times slowe

**Comment by Nyr** [comment]
> so is difference between windows PPTP and linux openvpn?

Yes, they are different protocols which have a different amount of overhead and which are prioritized differently by network operators. It's just different.

PPTP is obsolete and insecure, it shouldn't be used either way.

**Comment by lukebarone** [comment]
As well, since OpenVPN uses a higher quality of encryption, it takes more
computing time to encrypt and decrypt traffic, which slows down speeds as
well. I must agree with Nyr that the problem is likely related to the VPS
not being close enough, or having enough resources dedicated to your
instance.

**Comment by yakamasix** [comment]
what you can suggest me ? any solution? i dont need a secure connection, only used for betfair exchange, i must have a vpn tunnel because i like using oracle systems, because i have 3 accounts, i need 3 diffrent virtual system with 3 diffrent IPs, thats down speed is not really enaug for 5-6 tennis market opened similitianous.

**Comment by yakamasix** [comment]
i trying create L2TP connection but it doesnt work at all, it seems windows need any certificate, theres one more what i can do is create PPTP connection. I really understand you, but i looking for solution almost one week, theres many tutorials that doesnt work, unfortunatelly im a noob user linux :(

**Comment by lukebarone** [comment]
I'm not sure, as I don't understand what your requirements are. Most
solutions involving running scripts require a good knowledge of the
networking stacks, networking schemas, and shell commands. This script is
meant as a script for the road warriors, which does not sound like what
you're looking for. You can try posting most information on a help site,
like Super User, but be aware they expect a high-quality question to get a
good answer. See [this page](
http://www.dummies.com/careers/find-a-job/interviews/ten-tips-for-asking-good-questions/)
for a better idea of what works and doesn't for a good question.

**Comment by Nyr** [comment]
> what you can suggest me ? 

If you are in Poland and just want a foreign VPS, Germany will generally provide the best latency and bandwidth for you. I can even suggest a [provider](https://www.ultravps.eu/) if you want to try.

Other than that, we have a bit of trouble understanding your needs, but this is starting to get really out of the scope of this issue tracker so I'm locking the thread here. Please do not open a new issue here about the same matter.
